examples/example_piecewise_target_function.py

```diff
@@ -0,0 +1,185 @@
+
+    _def_output = "x_1 if x_0 > 0 else x_2"
+    _def_numpy_output = "np.piecewise(x_0, [x_0> 0, x_0<=0], [x_1 [x_0> 0] , x_2[x_0<=0]])"
+    _def_sympy_output = "Piecewise((x_1, x_0 > 0), (x_2, x_0 <= 0))"
+    _def_torch_output = "torch.where(x_0 > 0, x_1, x_2)"
+
+
+# %%
+# We define a piecewise target function.  The function applies different
+# transformations to the input depending whether the input is less or greater
+# than zero. Thus to achieve high fitness,
+# an individual must make use of the if/else operator.
+def f_target(x):
+    return np.select([x <= 0, x > 0], [-x, x ** 2 + 1.0])
+
+
+# %%
+# Then we define the objective function for the evolution. It uses the
+# mean-squared error between the output of the expression represented by a given
+# individual and the target function evaluated on a set of pseudo-random points.
+def objective(individual, target_function, rng):
+    """Objective function of the regression task.
+
+    Parameters
+    ----------
+    individual : Individual
+        Individual of the Cartesian Genetic Programming Framework.
+    target_function : Callable
+        Target function.
+    rng: numpy.random.RandomState
+
+    Returns
+    -------
+    Individual
+        Modified individual with updated fitness value.
+    """
+    if individual.fitness is not None:
+        return individual
+
+    n_function_evaluations = 1000
+
+    f_graph = individual.to_func()
+    y = np.empty(n_function_evaluations)
+    x = rng.uniform(-5, 5, size=n_function_evaluations)
+    for i, x_i in enumerate(x):
+        y[i] = f_graph([x_i])[0]
+
+    loss = np.mean((target_function(x) - y) ** 2)
+    individual.fitness = -loss
+
+    return individual
+
+
+seed = 999  # Todo: find better seed
+rng = np.random.RandomState(seed=seed)
+
+
+# %%
+# Next, we set up the evolutionary search. First, we define the parameters for
+# the population, the genomes of individuals, and the evolutionary
+# algorithm.
+population_params = {"n_parents": 1, "mutation_rate": 0.005, "seed": 8188211}
+
+genome_params = {
+    "n_inputs": 1,
+    "n_outputs": 1,
+    "n_columns": 20,
+    "n_rows": 1,
+    "levels_back": None,
+    "primitives": (
+        IfElseOperator,
+        cgp.Mul,
+        cgp.Add,
+        cgp.Sub,
+        cgp.ConstantFloat,
+    ),
+}
+
+ea_params = {"n_offsprings": 4, "tournament_size": 2, "n_processes": 2}
+
+evolve_params = {"max_generations": int(args["--max-generations"]), "min_fitness": 0.0}
+
+# create population that will be evolved
+pop = cgp.Population(**population_params, genome_params=genome_params)
+
+# create instance of evolutionary algorithm
+ea = cgp.ea.MuPlusLambda(**ea_params)
+
+# define callback for recording of fitness over generations
+history = {}
+history["fitness_parents"] = []
+history["champion_sympy"] = []
+
+
+def recording_callback(pop):
+    history["fitness_parents"].append(pop.fitness_parents())
+    try:
+        sympy_expression = pop.champion.to_sympy()
+    except TypeError:
```

Comment:
why is this necessary?

Reply:
`simpy.simplify` throws an error when `Piecewise` has the same expression for multiple conditions. This happens sometimes in the early generations of the algorithm.

Reply:
i see. is there any way to avoid this without the try-except construct? i guess this will also happen in user code, so it would be nice to catch these problems within the library? maybe we could move this try except to the `to_sympy` function and emit a warning if it's not possible?

Reply:
At the moment 
https://github.com/Happy-Algorithms-League/hal-cgp/blob/90afc8dcadcff8366b4f3641c47ea494f126bc51/cgp/individual.py#L179

Defaults to 'true', so the 'CartesianGraph'  instance simplify unless the user specifies for the individual not to. An option would be do move this try structure into 'CartesianGraph.to_sympy'. I will create a seperate PR with this

Reply:
sounds good. but we really should emit a warning, to make the user aware that something went wrong